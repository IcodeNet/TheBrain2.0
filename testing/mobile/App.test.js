/* eslint-env jest */
import { startAppMobileEnzyme } from './startAppMobileEnzyme'
import { LandingView } from '../../mobile/tests/e2e/viewObjects/LandingView'
import { CourseSelector } from '../../mobile/tests/e2e/viewObjects/CourseSelector'
import {
  getCoursesRepoWithDefaults,
  getFlashcardsRepoWithDefaults,
  getLessonsRepoWithDefaults
} from '../common/serverStateHelpers/helpers/reposWithDefaults'
import { LectureView } from '../../mobile/tests/e2e/viewObjects/LectureView'
import { QuestionsView } from '../../mobile/tests/e2e/viewObjects/QuestionsView'

const returnContext = async function () {
  const coursesRepository = await getCoursesRepoWithDefaults()
  const lessonsRepository = await getLessonsRepoWithDefaults()
  const flashcardsRepository = await getFlashcardsRepoWithDefaults()

  const context = {
    Courses: coursesRepository,
    Lessons: lessonsRepository,
    Flashcards: flashcardsRepository,
    req: {
      logIn: (user) => {
        console.log('Gandecki user', user)
        context.user = user
      }
    }
  }
  return context
}

jest.setTimeout(30000)

function timeout (ms) {
  return new Promise(resolve => setTimeout(resolve, ms))
}

describe('test', async () => {
  let context
  let driver
  beforeEach(async () => {
    context = await returnContext()
    driver = await startAppMobileEnzyme(context)
  })
  afterEach(() => {
    driver.wrapper.unmount()
  })
  it('renders the whole app without crashing', async () => {
    const landingView = new LandingView(driver)
    await landingView.assertIsVisible()
    await landingView.skipTutorial()
    //
    const courseSelector = new CourseSelector(driver)
    await courseSelector.assertIsVisible()
    await courseSelector.chooseBiology()
    //
    const lectureView = new LectureView(driver)

    await lectureView.assertIsVisible()
    await lectureView.skipLecture()
    //
    const questionsView = new QuestionsView(driver)
    await questionsView.hardcoreWarning.showHardcoreQuestions()
    //
    await driver.refresh()
    await questionsView.flashcard.assertQuestionShown('What is the name of this course')
    await questionsView.flashcard.flipFlashcard()

    await timeout(500) // TODO ugly hack, but will have to do for now
    await questionsView.flashcard.assertFlashcardShown('Biology')
  })
})
