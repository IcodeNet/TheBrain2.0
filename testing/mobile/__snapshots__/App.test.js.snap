// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the whole app without crashing 1`] = `"<rctsafeareaview style=\\"background-color: transparent;\\"><view style=\\"background-color: rgb(144, 80, 186); height: 100%;\\"><view></view><view><view style=\\"position: relative; width: 100%; height: 100%;\\"><view style=\\"top: 0px; z-index: 1000; width: 100%;\\"><view><view style=\\"width: 90%; opacity: 1;\\"><image resizemode=\\"contain\\" source=\\"1\\"></image></view><view style=\\"width: 35px; height: 35px;\\"><view style=\\"height: 3px; margin-left: 0px; width: 25px; margin-bottom: 0px; background-color: rgb(98, 196, 108);\\"></view><view style=\\"height: 3px; width: 25px; opacity: 1; background-color: rgb(98, 196, 108); margin-top: 4px;\\"></view><view style=\\"height: 3px; margin-left: 0px; width: 25px; background-color: rgb(98, 196, 108); margin-top: 4px;\\"></view></view></view></view><view style=\\"margin: 0px; height: 116px; width: 100%;\\"></view><view><view><text ellipsizemode=\\"tail\\">Choose a course:</text></view><view><view style=\\"elevation: 100px; width: 45%;\\"><view><view><view style=\\"margin: 10px; width: 120px; height: 120px; border-radius: 60px; border-width: 2px; border-color: white; opacity: 1;\\"><view style=\\"width: 100%; height: 100%; position: absolute; opacity: 1;\\"><image source=\\"1\\" style=\\"width: 40px; height: 80px;\\"></image></view></view><view style=\\"position: absolute; width: 136px; height: 136px;\\"><view style=\\"position: absolute; top: 10px; left: 68px; background-color: white; width: 0px; height: 0px; border-radius: 10px;\\"></view><view style=\\"position: absolute; top: 68px; left: 126px; background-color: white; width: 0px; height: 0px; border-radius: 10px;\\"></view><view style=\\"position: absolute; top: 126px; left: 68px; background-color: white; width: 0px; height: 0px; border-radius: 10px;\\"></view><view style=\\"position: absolute; top: 68px; left: 10px; background-color: white; width: 0px; height: 0px; border-radius: 10px;\\"></view></view></view></view><view style=\\"margin-bottom: 20px;\\"><text style=\\"color: white; text-align: center; font-size: 16px; opacity: 1;\\" ellipsizemode=\\"tail\\">Chemistry</text></view></view><view style=\\"elevation: 100px; width: 45%;\\"><view><view><view style=\\"margin: 10px; width: 120px; height: 120px; border-radius: 60px; border-width: 2px; border-color: white; opacity: 1;\\"><view style=\\"width: 100%; height: 100%; position: absolute; opacity: 1;\\"><image source=\\"1\\" style=\\"width: 35px; height: 70px;\\"></image></view></view><view style=\\"position: absolute; width: 136px; height: 136px;\\"><view style=\\"position: absolute; top: 10px; left: 68px; background-color: white; width: 0px; height: 0px; border-radius: 10px;\\"></view><view style=\\"position: absolute; top: 68px; left: 126px; background-color: white; width: 0px; height: 0px; border-radius: 10px;\\"></view><view style=\\"position: absolute; top: 126px; left: 68px; background-color: white; width: 0px; height: 0px; border-radius: 10px;\\"></view><view style=\\"position: absolute; top: 68px; left: 10px; background-color: white; width: 0px; height: 0px; border-radius: 10px;\\"></view></view></view></view><view style=\\"margin-bottom: 20px;\\"><text style=\\"color: white; text-align: center; font-size: 16px; opacity: 1;\\" ellipsizemode=\\"tail\\">Biology</text></view></view></view></view></view></view></view></rctsafeareaview>"`;
